include/triton/Dialect/TritonInstrument/IR/TritonInstrumentOps.td

```diff
@@ -11,7 +11,19 @@ class TTI_Op<string mnemonic, list<Trait> traits = []> :
     Op<TritonInstrument_Dialect, mnemonic, traits> {
 }
 
-// Define an array of pointers to shared memory buffers
+def TTI_ExperimentalAssertInThreadOp : TTI_Op<"experimental_assert_in_thread", [Pure]> {
```

Comment:
Pure and no return means it's always dead?
How is that different than a normal assert?